doc/axiom.md

```diff
@@ -12,9 +12,11 @@ Internationalized Resource Identifiers (IRIs) and their subset Universal Resourc
 
 ### Full IRIs
 
-Examples: `<http://absolute-iri-example.com/>`, `<relative-iri>`
+Examples: `<http://absolute-iri-example.com/>`, `<relative-iri>`, `(java.net.URL. <http://foo.org>)`
 
-Full IRIs in Flint are written as strings of the form `<my-iri-string>`. The string inside the angle bracket pair can include any characters **except** for whitespace, `^`, `<`, `>`, `"`, `\`, `|`, or `` ` ``. Translating to SPARQL does not affect full IRIs.
+Full IRIs in Flint are written as one of the following:
+- Strings of the form `<my-iri-string>`. The string inside the angle bracket pair can include any characters **except** for whitespace, `^`, `<`, `>`, `"`, `\`, `|`, or `` ` ``. Translating to SPARQL does not affect full IRIs.
+- Instances of either `java.net.URL`, `java.net.URI` (in Clojure), or `js/URL` (in ClojureScript). The inner string must follow the above restrictions.
```

Comment:
Hmm, I thought `java.net.URL` and `java.net.URI` would be used interchangeably by users in this context, which is why I had both of them extend the `IRI` protocol, but that might not always be the case?

Reply:
`java.net.URL` is a rather dangerous class; and people should IMHO avoid using it if they can.  Infact I'd urge you not to map it to anything at all.

The reason is because the [`.equals`]( https://docs.oracle.com/javase/7/docs/api/java/net/URL.html#equals(java.lang.Object)) method actually attempts to resolve the URL's to compare them for equality; i.e. it makes network requests!!!!

This bit me very hard a long time ago; where `java.net.URL` objects were used as keys in a HashMap; and we had to debug why the performance of our app tanked; we were unwittingly making hundreds of network requests, and it wasn't easy to discover that this was the reason why!

Reply:
Oh dear; when you said that my first thought was "that's a Log4Shell 2.0 just waiting to happen."

If that's the case, I'll go ahead and remove the `java.net.URL` extension; it would also have the benefit of making `java.net.URI` line up with `js/URI`.

Reply:
Removed in https://github.com/yetanalytics/flint/pull/27/commits/ccfbd473bd93e0835258ea8c1fdf140b13ea2dc9